internal/tiltfile/links.go

```diff
@@ -0,0 +1,93 @@
+package tiltfile
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+func convertLinks(val starlark.Value) ([]model.Link, error) {
+	// TODO: validate URL? (e.g. do we require scheme?)
+	if val == nil {
+		return nil, nil
+	}
+	switch val := val.(type) {
+	case starlark.String:
+		li, err := strToLink(val)
+		if err != nil {
+			return nil, err
+		}
+		return []model.Link{li}, nil
+	case link:
+		return []model.Link{val.Link}, nil
+	case starlark.Sequence:
+		var result []model.Link
+		it := val.Iterate()
+		defer it.Done()
+		var v starlark.Value
+		for it.Next(&v) {
+			switch v := v.(type) {
+			case starlark.String:
+				li, err := strToLink(v)
+				if err != nil {
+					return nil, err
+				}
+				result = append(result, li)
+			case link:
+				result = append(result, v.Link)
+			default:
+				return nil, fmt.Errorf("`link` arg %v includes element %v which must be a string or a link; is a %T", val, v, v)
+			}
+		}
+		return result, nil
+	default:
+		return nil, fmt.Errorf("`link` value must be an string, a link, or a sequence of those; is a %T", val)
+	}
+}
+
+func (s *tiltfileState) link(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var url, name string
+
+	if err := s.unpackArgs(fn.Name(), args, kwargs,
+		"url", &url,
+		"name?", &name); err != nil {
+		return nil, err
+	}
+
+	return link{
+		model.Link{
+			URL:  url,
+			Name: name,
+		},
+	}, nil
+}
+
+type link struct {
```

Comment:
should we embed a Starlark Struct for this?
https://pkg.go.dev/go.starlark.net/starlarkstruct#Struct

e.g.,

```
type link struct {
  Struct
  
  model.Link
}
```

Struct comes out of the box with a bunch of semantics that make sense for this value, rather than hand-implementing Truth(), Hash(), etc yourself.

Reply:
ooh neat, I was just cribbing from `portForward` that sgtm

Reply:
hm do we have an example of this somewhere in the code? (I can't see one.) i'm running into some trouble with pointer receivers and i'm going to leave this for now.

Reply:
sgtm, i can poke at it later